wallet/lnd/lnd.go

```diff
@@ -1,37 +1,35 @@
+
 package lnd
 
 import (
-	"bytes"
 	"context"
-	"encoding/json"
 	"encoding/hex"
 	"fmt"
-	"io"
-	"net/http"
+	"time"
 	"io/ioutil"
-	"github.com/sulusolutions/gol402/wallet"
-)
-
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
 
-type lndWalletResponse struct {
-	PaymentError    string `json:"payment_error"`
-	PaymentPreimage string `json:"payment_preimage"`
-	PaymentRoute    struct{} `json:"payment_route"`
-	PaymentHash     string `json:"payment_hash"`
-}
+	"github.com/lightninglabs/lndclient"
+	"github.com/lightningnetwork/lnd/lnrpc"
+)
 
+	// Define payment request
+	type CustomSendPaymentRequest struct {
+		Invoice          string
+		MaxFee           int64
+		Timeout          time.Duration
+		AllowSelfPayment bool
+	}
 
-// LNDWallet implements the Wallet interface using the LND WALLET REST API.
 type LNDWallet struct {
-	// BaseURL is the base URL for the  LND wallet API.
 	BaseURL string
```

Comment:
Since we are using gRPC we do not really need a BaseURL, which would be more common with REST APIs. With gRPC we usually deal with an address.